### ISSUE TITLE
[Bricks Builder] [jetEngine] Dynamic Visibility Condition

### ISSUE THREAD

**Issue by sekhsoyebali** [CrocoBlock/suggestions#6962]
There are only a limited number of Dynamic Conditions available in Bricks Builder
For example, the following conditions are not available
Query has item
Is profile page
Is mobile device
Listing is even, is odd
Value is checked
Switcher condition
It would be helpful if JetEngine's Dynamic Visibility Conditions could be added.

Good News: Just now, Bricks has released a Conditions API in version 1.8.4.

https://academy.bricksbuilder.io/article/element-conditions/#api

**Comment by bjoernzosel** [comment]
well it's great news for developers, but not for non-coders...

**Comment by ghost** [comment]
Hi Crocoblock Devs, any updates for this topic?
Sincerely Edgar

**Comment by sekhsoyebali** [comment]
> Hi Crocoblock Devs, any updates for this topic? Sincerely Edgar

Yes Crocoblock, please update us!

**Comment by breakused** [comment]
any news? :)

**Comment by ghost** [comment]
Yes, this is so important. All the other plugins by you that are not fully yet supported for Bricks, like JetElements are not that important. But to create a basic to advanced website I just miss  a few options for the element render conditions inside Bricks. Lost of stuff does work already, but not all of it. For example: I just cannot check whether dynamic data is set or not, they just offer the "<", ">", "==", "!==", ... operators. For me, these options are missing, that are crucial:

Missing data sources: (plus their negative versions)

- Query results count (I know, Bricks does already offer {query_results_count}), but it's meant for their query loop feature)
- In the list
- Exists - MOST IMPORTANT, I CAN'T PROPERLY WORK WITHOUT IT :(
- Between
- Regexp
- Value is checked
- Is post by queried user
- Single post type is
- Post type archive is
- Taxonomy archive is
- Is current post author
- Post has terms
- Is parent post
- Is child post of
- Time period
- Is even listing item
- Is odd listing item
- Is N listing item

- Macros dynamic tag

- Suggestion: Is post ID of listing item
- Suggestion: Is first listing item
- Suggestion: Is last listing item

And please offer these methods to repeater meta fields (currently only supported for regular ones), so that we have more control for repeater listings. Currently thats just a pain..

**Comment by breakused** [comment]
There is one Workaround if needed. That was introduced by itchycode. Thanks to him for this.

![itchy](https://github.com/Crocoblock/suggestions/assets/37920259/bd261b19-c19c-400b-af95-60ca0892135f)

[https://itchycode.com/create-my-custom-conditional-visibility-feature-in-bricks-theme-without-add-on/](https://itchycode.com/create-my-custom-conditional-visibility-feature-in-bricks-theme-without-add-on/)

**Comment by Cobasites** [comment]
Looking for the same, dynamic visibility / conditions are so powerful, and this is the most important thing missing when switching from Elementor to Bricks Builder.

**Comment by stefanradu9** [comment]
Any updates ?

**Comment by Crocoblock** [comment]
The JetEngine 3.6.4 update added the ability to use conditions via shortcode

Now you can generate conditions in the Engine using shortcodes https://i.imgur.com/RafABAs.png 
-The first type is Selfclosing, returns true or false (will be used for Bricks as well)
-The second type - Enclosing, returns the content that is between the shortcode tags

So you can generate a shortcode and use it in the Bricks conditions 
https://prnt.sc/n6SlJVzBLWe6 https://prnt.sc/MDfVa9oxqYt2 https://prnt.sc/nkBl-0hkT-Mo 
result: https://prnt.sc/EmLkPinwBxoN
